openslides_backend/action/actions/user/base_json_upload.py

```diff
@@ -138,18 +143,22 @@ def validate_entry(self, entry: dict[str, Any]) -> dict[str, Any]:
 
                 self.row_state = ImportState.DONE
                 entry["id"] = id_
-                entry["username"] = {
-                    "value": username,
-                    "info": ImportState.DONE,
-                    "id": id_,
-                }
+                if isinstance(username, str):
+                    entry["username"] = {
+                        "value": username,
+                        "info": ImportState.DONE,
+                        "id": id_,
+                    }
             elif check_result == ResultType.NOT_FOUND or id_ == 0:
                 self.row_state = ImportState.NEW
         else:
-            if not (entry.get("first_name") or entry.get("last_name")):
+            if not entry.get("username") or (
+                not isinstance(entry.get("username"), str)
+                and not entry.get("username", {}).get("value")
+            ):
                 self.row_state = ImportState.ERROR
                 messages.append(
-                    "Cannot generate username. Missing one of first_name, last_name."
+                    "Cannot generate username. Missing one of first_name, last_name or a unique member_number."
```

Comment:
It seems that at this point of code you still not tried to look for the existence of a member_number, see line 203